src/components/CreatePhotoGallery.jsx

```diff
@@ -0,0 +1,361 @@
+// TODO: Remove main error display after delay
+
+import React, { Component } from 'react';
+import Dropzone from 'react-dropzone';
+import * as firebase from 'firebase';
+import FormInput from './FormInput';
+import ErrorMessages from './ErrorMessages';
+import { fieldValidationsPhotoGallery, run, ruleRunner, required } from '../utils/index';
+
+let localFieldValidations = [...fieldValidationsPhotoGallery];
+
+const createFieldValidations = (file) => {
+  localFieldValidations.push(
+    ruleRunner(file.name, 'Alt text', required),
+  );
+};
+
+const PreviewGrid = require('react-packery-component')(React);
+
+const packeryOptions = {
+  gutter: 10,
+  itemSelector: '.preview-item',
+};
+
+class CreatePhotoGallery extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      galleryName: '',
+      images: [],
+      unsavedChanges: false,
+      uploadProgress: {
+        files: [],
+        totalBytes: 0,
+        bytesTransferred: 0,
+        percentProgress: 0,
+      },
+      uploading: false,
+      mainErrorDisplay: '',
+      showErrors: {
+        galleryName: false,
+      },
+      validationErrors: {},
+      touched: {
+        galleryName: false,
+      },
+      submit: false,
+      componentRef: null,
+    };
+
+    this.onImageDrop = this.onImageDrop.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.removeFile = this.removeFile.bind(this);
+    this.handleBlur = this.handleBlur.bind(this);
+    this.errorFor = this.errorFor.bind(this);
+    this.handleFocus = this.handleFocus.bind(this);
+  }
+
+  onImageDrop(files) {
+    // check to ensure file names are unique
+    if (files.some(file => this.state.images.some(image => image.name === file.name))) {
+      this.displayMainError('File names must be unique');
+      return;
+    }
+
+    files.forEach((file) => {
+      // create a new ruleRunner for each new image
+      createFieldValidations(file);
+
+      this.setState({
+        images: [...this.state.images, ...files],
+        validationErrors: run(Object.assign({}, this.state), localFieldValidations),
```

Comment:
object  spread operator in one line and `Object.assign` in the line below, since they accomplish the same thing it would be cool to have them be consistent.